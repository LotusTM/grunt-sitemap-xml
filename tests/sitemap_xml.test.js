/* eslint-env jest */

const { runGrunt, grunt } = require('./utils/grunt')
const { config, file: { read } } = grunt

describe('sitemap_xml task', () => {
  beforeAll(() => runGrunt(['test']))

  describe('with default settings', () => {
    it('should produce correct xml file', () => {
      expect(read(config('path.defaultOptionsMap'))).toMatchSnapshot()
    })
  })

  describe('with custom settings', () => {
    it('should produce correct xml file', () => {
      expect(read(config('path.customOptionsMap'))).toMatchSnapshot()
    })
  })
})
