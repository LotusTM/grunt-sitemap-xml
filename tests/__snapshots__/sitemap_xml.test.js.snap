// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sitemap_xml task with custom settings should produce correct xml file 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\" xsi:schemaLocation=\\"http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd\\">
  <url>
    <loc>true/pages/blog.html</loc>
    <lastmod>2017-09-28</lastmod>
    <changefreq>monthly</changefreq>
    <priority>0.1</priority>
  </url>
  <url>
    <loc>true/pages/index.html</loc>
    <lastmod>2017-09-28</lastmod>
    <changefreq>monthly</changefreq>
    <priority>0.1</priority>
  </url>
  <url>
    <loc>true/pages/news.html</loc>
    <lastmod>2017-09-28</lastmod>
    <changefreq>monthly</changefreq>
    <priority>0.1</priority>
  </url>
  <url>
    <loc>true/pages/short.htm</loc>
    <lastmod>2017-09-28</lastmod>
    <changefreq>monthly</changefreq>
    <priority>0.1</priority>
  </url>
</urlset>"
`;

exports[`sitemap_xml task with default settings should produce correct xml file 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?><urlset xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\" xmlns:xsi=\\"http://www.w3.org/2001/XMLSchema-instance\\" xsi:schemaLocation=\\"http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd\\"><url><loc>https://github.com/LotusTM/grunt-sitemap-xml/pages/blog.html</loc><lastmod>2017-09-28T14:28:33+03:00</lastmod><changefreq>weekly</changefreq><priority>0.5</priority></url><url><loc>https://github.com/LotusTM/grunt-sitemap-xml/pages/</loc><lastmod>2017-09-28T14:28:33+03:00</lastmod><changefreq>weekly</changefreq><priority>0.5</priority></url><url><loc>https://github.com/LotusTM/grunt-sitemap-xml/pages/news.html</loc><lastmod>2017-09-28T14:28:33+03:00</lastmod><changefreq>weekly</changefreq><priority>0.5</priority></url></urlset>"`;
